solidity/test/Polymer7683.t.sol

```diff
@@ -0,0 +1,392 @@
+
+contract Polymer7683Test is Test {
+    using TypeCasts for address;
+
+    Polymer7683 polymer7683;
+    MockCrossL2Prover prover;
+    
+    address owner;
+    address permit2;
+    uint256 localChainId;
+    uint256 destChainId;
+    address destContract;
+
+    function setUp() public {
+        owner = makeAddr("owner");
+        permit2 = makeAddr("permit2");
+        localChainId = 1;
+        destChainId = 2;
+        destContract = makeAddr("destContract");
+
+        prover = new MockCrossL2Prover();
+        
+        vm.prank(owner);
+        polymer7683 = new Polymer7683(
+            ICrossL2ProverV2(address(prover)),
+            permit2,
+            localChainId
+        );
+    }
+
+    function _createSettlementProof(
+        uint256 chainId,
+        address emitter,
+        bytes32 orderId,
+        bytes memory fillerData
+    ) internal returns (bytes memory) {
+        OrderData memory orderData = OrderData({
+            sender: TypeCasts.addressToBytes32(address(this)),
+            recipient: TypeCasts.addressToBytes32(address(this)),
+            inputToken: bytes32(0),
+            outputToken: bytes32(0),
+            amountIn: 1e18,
+            amountOut: 2e18,
+            senderNonce: 0,
+            originDomain: uint32(localChainId),
+            destinationDomain: uint32(chainId),
+            destinationSettler: TypeCasts.addressToBytes32(emitter),
+            fillDeadline: uint32(block.timestamp + 1 hours),
+            data: ""
+            });
+
+        bytes memory originData = OrderEncoder.encode(orderData);
+    
+        // Create the proof data
+        bytes memory topics = abi.encodePacked(
+            keccak256("Filled(bytes32,bytes,bytes)"),
+            bytes32(chainId)
+        );
+    
+        bytes memory data = abi.encode(orderId, originData, fillerData);
+    
+        prover.setExpectedEvent(
+            uint32(chainId),
+            emitter,
+            topics,
+            data
+        );
+    
+        return "dummy_proof";
+    }
+
+    function _createRefundProof(
+        uint256 chainId,
+        address emitter,
+        bytes32[] memory orderIds
+    ) internal returns (bytes memory) {
+        bytes32[] memory topicsArray = new bytes32[](1);
+        topicsArray[0] = keccak256("Refund(bytes32[])");
+        bytes memory topics = abi.encode(topicsArray);
+
+        bytes memory data = abi.encode(orderIds);
+    
+        prover.setExpectedEvent(
+            uint32(chainId),
+            emitter,
+            topics,
+            data
+        );
+    
+        return "dummy_proof";
+    }
+
+    function test_setDestinationContract() public {
+        vm.prank(owner);
+    
+        polymer7683.setDestinationContract(destChainId, destContract);
+        address stored = polymer7683.destinationContracts(destChainId);
+        assertEq(stored, destContract, "Destination contract address mismatch");
+    }
+
+    function test_handleSettlementWithProof_success() public {
```

Comment:
I suspect this test is not working as expected, because the order hasn't been open
I would assert that the `_handleSettleOrder` works. Provably asserting that the `Settled` is emitted.

Reply:
thanks for catching this!
I added open order logic to this test and add assertion to expect the event to be emitted.